examples/c/7_series/gnss_ins/7_series_gnss_ins_example.c

```diff
@@ -0,0 +1,1341 @@
+///             - Fix information
+///
+/// @param _device Pointer to the initialized MIP device interface
+///
+void configure_gnss_message_format(mip_interface* _device)
+{
+    // Note: Querying the device base rate is only one way to calculate the descriptor decimation
+    // We could have also set it directly with information from the datasheet
+
+    MICROSTRAIN_LOG_INFO("Getting the base rate for GNSS 1 data.\n");
+    uint16_t       gnss_base_rate_1;
+    mip_cmd_result cmd_result = mip_3dm_get_base_rate(
+        _device,
+        MIP_GNSS1_DATA_DESC_SET, // Data descriptor set
+        &gnss_base_rate_1        // Base rate out
+    );
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not get GNSS 1 base rate!\n");
+    }
+
+    // Supported sample rates can be any value from 1 up to the base rate
+    // Note: Decimation can be anything from 1 to 65,565 (uint16_t::max)
+    if (SAMPLE_RATE_HZ == 0 || SAMPLE_RATE_HZ > gnss_base_rate_1)
+    {
+        command_failure_terminate(
+            _device,
+            MIP_NACK_INVALID_PARAM,
+            "Invalid sample rate of %dHz! Supported rates are [1, %d].\n",
+            SAMPLE_RATE_HZ,
+            gnss_base_rate_1
+        );
+    }
+
+    // Calculate the decimation (stream rate) for the device based on its base rate
+    const uint16_t gnss_decimation_1 = gnss_base_rate_1 / SAMPLE_RATE_HZ;
+    MICROSTRAIN_LOG_INFO("Decimating GNSS 1 base rate %d by %d to stream data at %dHz.\n",
+        gnss_base_rate_1,
+        gnss_decimation_1,
+        SAMPLE_RATE_HZ
+    );
+
+    // Descriptor rate is a pair of data descriptor set and decimation
+    const mip_descriptor_rate gnss_descriptors_1[1] = {
+        { MIP_DATA_DESC_GNSS_FIX_INFO, gnss_decimation_1 }
+    };
+
+    MICROSTRAIN_LOG_INFO("Configuring message format for GNSS 1 data.\n");
+    cmd_result = mip_3dm_write_message_format(
+        _device,
+        MIP_GNSS1_DATA_DESC_SET,                                    // Data descriptor set
+        sizeof(gnss_descriptors_1) / sizeof(gnss_descriptors_1[0]), // Number of descriptors to include
+        gnss_descriptors_1                                          // Descriptor array
+    );
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not set message format for GNSS 1 data!\n");
+    }
+
+    // Note: Typically, the base rates for all GNSS descriptors will be the same, but adding this for completeness
+    MICROSTRAIN_LOG_INFO("Getting the base rate for GNSS 2 data.\n");
+    uint16_t gnss_base_rate_2;
+    cmd_result = mip_3dm_get_base_rate(
+        _device,
+        MIP_GNSS2_DATA_DESC_SET, // Data descriptor set
+        &gnss_base_rate_2        // Base rate out
+    );
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not get GNSS 2 base rate!\n");
+    }
+
```

Comment:
This duplicated code is confusing. Make this function set up only one receiver, then call it twice with appropriate parameters for each receiver.